python/libwholegraph/CMakeLists.txt

```
@@ -0,0 +1,37 @@
+# =============================================================================
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.30.4 FATAL_ERROR)
+
+include(../../cmake/rapids_config.cmake)
+include(rapids-cmake)
+include(rapids-cuda)
+include(rapids-cpm)
+
+rapids_cuda_init_architectures(LIBWHOLEGRAPH)
+
+project(
+  LIBWHOLEGRAPH
+  VERSION "${RAPIDS_VERSION}"
+  LANGUAGES CXX CUDA
+)
+
+rapids_cpm_init()
```

```suggestion
include(../../cmake/rapids_config.cmake)
include(rapids-cmake)
include(rapids-cuda)

rapids_cuda_init_architectures(LIBWHOLEGRAPH)

project(
  LIBWHOLEGRAPH
  VERSION "${RAPIDS_VERSION}"
  LANGUAGES CXX CUDA
)
```

I think the CPM stuff can be removed here. `rapids_cpm_init()` will be called here, in the project you've added via `add_subdirectory()`:

https://github.com/rapidsai/cugraph-gnn/blob/3d67322d679c8c08e7bd973279db378dff483058/cpp/CMakeLists.txt#L113

You'd only need to call `rapids_cpm_init()` up here in `libwholegraph/CMakeLists.txt` if you were invoking any CPM codepaths directly from here, which you aren't.